includes/js/komoju-checkout-blocks.js

```diff
@@ -0,0 +1,102 @@
+const { useEffect, useCallback, useRef, createElement } = window.wp.element;
+
+function registerPaymentMethod(paymentMethod) {
+    let name = `${paymentMethod.id}`
+    const settings = window.wc.wcSettings.getSetting(`${name}_data`, {});
+
+    const komojuFields = createElement('komoju-fields', {
+        'token': '',
+        'name': 'komoju_payment_token',
+        'komoju-api': settings.komojuApi,
+        'publishable-key': settings.publishableKey,
+        'session': settings.session,
+        'payment-type': settings.paymentType,
+        style: { display: 'none' },
+    });
+
+    const label = createElement('div', {
+        style: { display: 'block', alignItems: 'center', justifyContent: 'center' }
+    },
+        window.wp.htmlEntities.decodeEntities(settings.title || window.wp.i18n.__('NULL GATEWAY', 'test_komoju_gateway')),
+        createElement('img', {
+            src: settings.icon,
+            alt: settings.title || 'Payment Method Icon',
+            style: { display: 'flex', alignItems: 'center', justifyContent: 'center', marginLeft: '10px' }
+        }),
+        komojuFields
+    );
+
+    const KomojuComponent = ({ activePaymentMethod, emitResponse, eventRegistration }) => {
+        const { onPaymentSetup } = eventRegistration;
+
+        useEffect(() => {
+            const komojuField = document.querySelector(`komoju-fields[payment-type='${paymentMethod.paymentType}']`);
+            komojuField.style.display = 'block';
+
+            const unsubscribe = onPaymentSetup(async () => {
+                console.log('onPaymentSetup', paymentMethod.id, activePaymentMethod)
+                if (paymentMethod.id != activePaymentMethod) return;
+                // Exceptions
+                if (paymentMethod.id === 'komoju_paidy') return;
+                if (paymentMethod.id === 'komoju_net_cash') return;
+                if (paymentMethod.id === 'komoju_bit_cash') return;
+                if (paymentMethod.id === 'komoju_web_money') return;
+                if (paymentMethod.id === 'komoju_pay_easy') return;
```

Comment:
@Dinwy We should invert the logic here. We probably want an array for payment methods which support KOMOJU fields in case we introduce more payment methods in the future.

Reply:
Sure thing. This is fixed now. [Commit](https://github.com/degica/komoju-woocommerce/pull/111/commits/a57548154f5d66e4f828c6294768a9a98a9c3818)